java/src/test/java/org/hyperledger/fabric/client/CommonCheckpointerTest.java

```diff
@@ -0,0 +1,78 @@
+package org.hyperledger.fabric.client;
+
+import com.google.protobuf.ByteString;
+import org.hyperledger.fabric.protos.peer.ChaincodeEventPackage;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public abstract  class CommonCheckpointerTest {
+
+    Checkpointer checkpointerInstance;
+
+    void assertData(Checkpointer checkpointer, long blockNumber, Optional<String> transactionId) {
+        assertThat(checkpointer.getBlockNumber()).isEqualTo(blockNumber);
+        if (transactionId.isPresent()) {
+            assertTrue(checkpointer.getTransactionId().isPresent());
+            assertThat(checkpointer.getTransactionId().get()).isEqualTo(transactionId.get());
+        } else {
+            assertFalse(checkpointer.getTransactionId().isPresent());
+        }
+    }
```

Comment:
I wouldn't pass an Optional in here. AssertJ has better ways of checking an Optional value too. If you want to have assert methods, I'd probably break it up like below with method overloading or two differently named methods, or you might find simpler assertions make it not worth the effort. If sub-classess will use these methods, make them `protected`; otherwise `private`.
```suggestion
    protected assertCheckpoint(final Checkpoint checkpoint, final long blockNumber) {
        assertThat(checkpoint.getBlockNumber()).isEqualTo(blockNumber);
        assertThat(checkpoint.getTransactionId()).isNotPresent();
    }

    protected assertCheckpoint(final Checkpoint checkpoint, final long blockNumber, final String transactionId) {
        assertThat(checkpoint.getBlockNumber()).isEqualTo(blockNumber);
        assertThat(checkpoint.getTransactionId()).isEqualTo(transactionId);
    }
```

Reply:
In case of checkpointBlock , should we set  transactionID to "" or null ?
If empty string, then `checkpoint.getTransactionId().isNotPresent() `will fail